### PR TITLE
bump go version to 1.22.4

### PR DESCRIPTION
**Pull request by dhiaayachi**
Upgrade go CI to use 1.22.4

### CODE REVIEW

**Comment by dhiaayachi** [comment]
@DanStough CI has been failing for a while now. I tried to fix it but I don't really understand any of those tests and it will take me a long time. I will merge this as is (after your approval) and then we will decide what to do about those failing tests.

**Review by DanStough — APPROVED** [review]

**Pull request merged**
